app/assets/javascripts/index/history-changesets-layer.js

```diff
@@ -119,14 +119,12 @@ OSM.HistoryChangesetsLayer = L.FeatureGroup.extend({
     const changeset = this._changesets.get(id);
     if (!changeset) return;
 
-    let highlightRect = this._highlightLayer.getLayer(id);
-    if (!state && highlightRect) {
-      this._highlightLayer.removeLayer(highlightRect);
-    }
-    if (state && !highlightRect) {
-      highlightRect = L.rectangle(changeset.bounds, this._getHighlightStyle(changeset));
+    if (state) {
+      const highlightRect = L.rectangle(changeset.bounds, this._getHighlightStyle(changeset));
```

Comment:
So we're now relying on leaflet discard any pre-existing layer with the same ID when we add this one? Is that documented anywhere?

Or is the assumption just that we won't toggle a changeset on that is already on?

Reply:
We are relying on:
- if the highlight layer is already added, it has correct geometry, style etc
- Leaflet [won't add a layer with the same id to the group](https://github.com/Leaflet/Leaflet/blob/c4a1e362bfd7f7709efdaff94f154e100a706129/src/layer/FeatureGroup.js#L29-L31), but looks like it's not documented unless [this counts as documented](https://github.com/Leaflet/Leaflet/pull/798)

Reply:
Overriding `getLayerId` to check if the layer is added by id is also not documented, but @jfirebaugh made it work both here a56d1036d58d45777d6adeffd5a950519bc70318 and in Leaflet https://github.com/Leaflet/Leaflet/pull/1581.

Reply:
I guess that's probably good enough.